pkg/internal/mock/storage/mock_store.go

```diff
@@ -34,28 +36,30 @@ func (s *MockStoreProvider) Close() error {
 
 // MockStore mock store.
 type MockStore struct {
-	store map[string][]byte
-	lock  sync.RWMutex
+	Store  map[string][]byte
+	lock   sync.RWMutex
+	ErrPut error
+	ErrGet error
 }
 
 // Put stores the key and the record
 func (s *MockStore) Put(k string, v []byte) error {
 	s.lock.Lock()
-	s.store[k] = v
+	s.Store[k] = v
 	s.lock.Unlock()
 
-	return nil
+	return s.ErrPut
 }
 
 // Get fetches the record based on key
 func (s *MockStore) Get(k string) ([]byte, error) {
 	s.lock.RLock()
 	defer s.lock.RUnlock()
 
-	val, ok := s.store[k]
+	val, ok := s.Store[k]
 	if !ok {
-		return nil, fmt.Errorf("no value found for the key %s", k)
+		return nil, nil
```

Comment:
I believe we convened to not return nil, nil or !nil, !nil

am I wrong?

Reply:
it's mock